### PR TITLE
[AKS] Change subnetPrefix to subnetCIDR

### PR DESCRIPTION
**Pull request by akshaysngupta**
* This addon is right now in `Public Preview`.
* This PR changes parameter `--appgw-subnet-prefix` to `--appgw-subnet-cidr` in `az aks create` and `az aks enable-addons` commands.
* This change was suggested by AKS PMs.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by akshaysngupta** [comment]
@gtracer can you help review ?

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -260,9 +260,9 @@
         - name: --appgw-name
           type: string
           short-summary: Name of the application gateway to create/use in the node resource group. Use with ingress-azure addon.
-        - name: --appgw-subnet-prefix
+        - name: --appgw-subnet-cidr
```
Comment (arrownj) [inline src/aks-preview/azext_aks_preview/_help.py:263]: So this is a breaking change to existing users, right ?

File: src/aks-preview/azext_aks_preview/_params.py [hunk 2]
```diff
@@ -108,6 +108,11 @@ def load_arguments(self, _):
         c.argument('enable_managed_identity', action='store_true')
         c.argument('assign_identity', type=str, validator=validate_assign_identity)
         c.argument('disable_sgxquotehelper', action='store_true')
+        c.argument('appgw_name', options_list=['--appgw-name'])
```
Comment (arrownj) [inline src/aks-preview/azext_aks_preview/_params.py:111]: What's other parameters except `--appgw-subnet-cidr` for ? There is no introduction for them.

File: src/aks-preview/azext_aks_preview/_help.py (hunk 1 above)
Comment (akshaysngupta) [reply src/aks-preview/azext_aks_preview/_help.py:263]: yes, this is a breaking change for existing users of addon.

File: src/aks-preview/azext_aks_preview/_params.py (hunk 2 above)
Comment (akshaysngupta) [reply src/aks-preview/azext_aks_preview/_params.py:111]: these parameters were added before but I didn't make change in this file.

**Comment by arrownj** [comment]
Hi @zqingqing1 , could you please help review this change and involve more related people ? Thanks

**Comment by akshaysngupta** [comment]
@arrownj added support for both with deprecation message for old

File: src/aks-preview/azext_aks_preview/_params.py [hunk 3]
```diff
@@ -110,6 +110,12 @@ def load_arguments(self, _):
         c.argument('disable_sgxquotehelper', action='store_true')
         c.argument('kubelet_config', type=str)
         c.argument('linux_os_config', type=str)
+        c.argument('appgw_name', options_list=['--appgw-name'], arg_group='Application Gateway')
+        c.argument('appgw_subnet_prefix', options_list=['--appgw-subnet-prefix'], arg_group='Application Gateway', deprecate_info=c.deprecate(redirect='--appgw-subnet-cidr', hide=True))
+        c.argument('appgw_subnet_cidr', options_list=['--appgw-subnet-cidr'], arg_group='Application Gateway')
+        c.argument('appgw_id', options_list=['--appgw-id'], arg_group='Application Gateway')
+        c.argument('appgw_subnet_id', options_list=['--appgw-subnet-id'], arg_group='Application Gateway')
+        c.argument('appgw_watch_namespace', options_list=['--appgw-watch-namespace'], arg_group='Application Gateway')
```
Comment (fengzhou-msft) [inline src/aks-preview/azext_aks_preview/_params.py:118]: Just a kind reminder, CLI core will automatically convert `_` into `-` in argument name for the long option. If you don't add a short option or don't change the long option name, you can omit the `options_list` part for future PRs.

**Review by fengzhou-msft — APPROVED** [review]

**Comment by fengzhou-msft** [comment]
@akshaysngupta the version is not bumped, the change will not be released until next version bump. Just to confirm, is this by plan?

**Comment by akshaysngupta** [comment]
Should I bump the version in this PR?

**Comment by fengzhou-msft** [comment]
> Should I bump the version in this PR?

Yes if you plan to release the changes after the PR gets merged.

**Pull request merged**